src/idl_gen_cpp.cpp

```diff
@@ -2332,7 +2341,13 @@ class CppGenerator : public BaseGenerator {
         auto vector_type = field.value.type.VectorType();
         switch (vector_type.base_type) {
           case BASE_TYPE_STRING: {
-            code += "_fbb.CreateVectorOfStrings(" + value + ")";
+            // Use by-function serialization to emulate CreateVectorOfStrings();
+            // this works also with non-std strings.
+            code += "_fbb.CreateVector<flatbuffers::Offset<flatbuffers::String>> ";
```

Comment:
This is not great, as it creates more code for all users, even though it is needed for a tiny minority. Would be better to leave the original in there for most cases.

Reply:
Okay, now if type is std::string the normal CreateVectorOfStrings() is used, in all other cases the function.